pkg/controller/machine/machine_controller.go

```diff
@@ -550,7 +551,7 @@ func (r *Reconciler) deleteMachine(ctx context.Context, prov cloudprovidertypes.
 		return nil, nil
 	}
 
-	return nil, r.deleteNodeForMachine(ctx, machine)
+	return r.deleteNodeForMachine(ctx, machine)
```

Comment:
Just call this method here when the cloud provider is vsphere and the Volume attachment are gone.